LEGO1/mxwavepresenter.cpp

```diff
@@ -84,11 +87,46 @@ MxBool MxWavePresenter::FUN_100b1ba0()
 	return !m_unk65 || FUN_100b1b60() != m_unk64;
 }
 
-// OFFSET: LEGO1 0x100b1bd0 STUB
+// OFFSET: LEGO1 0x100b1bd0
 void MxWavePresenter::FUN_100b1bd0(void* p_audioPtr, MxU32 p_length)
 {
-	// Lock/Unlock on m_dsBuffer
-	// TODO
+	DWORD dwStatus;
+	LPVOID pvAudioPtr1;
+	DWORD dwOffset;
+	LPVOID pvAudioPtr2;
+	DWORD dwAudioBytes1;
+	DWORD dwAudioBytes2;
+
+	dwOffset = m_length * m_unk64;
+	m_dsBuffer->GetStatus(&dwStatus);
+
+	if (dwStatus == DSBSTATUS_BUFFERLOST) {
```

Comment:
Maybe this is a switch statement?

Reply:
Not sure what you mean, there is no switching on multiple values here.